app/components/view_component_reflex/component.rb

```diff
@@ -84,13 +84,8 @@ def can_render_to_string?
     def init_key
       # we want the erb file that renders the component. `caller` gives the file name,
       # and line number, which should be unique. We hash it to make it a nice number
-      erb_file = caller.select { |p| p.match? /.\.html\.(haml|erb|slim)/ }[1]
```

Comment:
This file is the file that renders the component, not the component template itself

Reply:
Is there a concern with this? I think it is a 1:1 association between .rb + the template file in VCRs, so it is ok to use the filepath of the .rb file in liue of the template file?

Reply:
So the way it works is if you have a view like this

```
# app/views/whateverindex.html.erb
<%= render MyComponent.new %>
<%= render MyComponent.new %>
<%= render MyComponent.new %>
```

Then we need a different key for all 3 of these components. 

The current solution as it stands gets the path to the `index.html.erb` (not the `my_component.html.erb`), and uses the line number to create distinct keys. 

So the strings we're hashing above are

```
app/views/whatever/index.html.erb:1
app/views/whatever/index.html.erb:2
app/views/wahtever/index.html.erb:3
```